docs/principles-for-package-repository-security.md

```diff
@@ -0,0 +1,105 @@
+
+### Authorization
+
+Applies to: package repositories that have user accounts and accept built packages.
+
+- Level 1
+  - The package repository allows maintainers to provision API keys scoped to specific packages. This allows maintainers to maintain packages via automated workflows without needing to provide their account password.
+- Level 2
+  - The package repository supports role-based access control (RBAC) for maintainers, allowing separate roles for managing users and publishing packages.
+- Level 3
+  - The package repository provisions short-lived identity tokens via  OpenID Connect (OIDC) to prevent needing to provision long-lived API keys. Example: [trusted publishers](https://docs.pypi.org/trusted-publishers/)
+  - The package repository API tokens are integrated into common third-party secret scanning programs like GitGuardian, GitHub, etc. This typically requires using a known prefix or pattern to reduce false-positives.
+  - The package repository supports providing [build provenance](https://repos.openssf.org/build-provenance-for-all-package-registries) for packages.
+
+### General Capabilities
+
+Applies to: all package repositories
+
+- Level 1
+  - The package repository has published a vulnerability disclosure policy allowing security researchers to report vulnerabilities to the package repository, and receive legal safe harbor when doing so. The package repository should strive to remediate reported vulnerabilities within 90 days.
+  - The package repository takes steps to prevent typosquatting attacks around package names. This may include namespacing, detecting similar names, or other actions.
+- Level 2
+  - The package repository has an unpublish policy to prevent specific versions of a package from being replaced, and, if a package is deleted, prevent others from reusing the same package name and version.
+  - The package repository allows users to report a package as suspicious or malicious via the registry UI, tooling CLI, and via API.
+  - The package repository takes steps to detect malware, including scanning for known malware code snippets and hashes.
+- Level 3
+  - Undergo periodic cloud security review (and optionally publish remediations once they are complete)
+  - The package repository publishes an event transparency log to enable detection of anomalous behaviors (e.g. [replicate.npmjs.com](https://replicate.npmjs.com/)).
+  - Malicious packages are published using OSV (e.g. [here](https://github.com/ossf/malicious-packages))
+
+### CLI Tooling
+
+Applies to: all package management ecosystems, although here we are primarily focused on the CLI tooling instead of the registry itself.
+
+- Level 1
+  - The CLI allows installing dependencies that are pinned based on hash, version, etc.
+- Level 2
+  - The package repository warns of known security vulnerabilities in dependencies, in both the package repository UI and CLI tools when installing.
+- Level 3
+  - The CLI has functionality to produce SBOMs.
```

Comment:
```suggestion
  - The CLI has functionality to produce SBOMs.
```

I think this could be expanded: produce SBOMs for what? (applications? libraries?) based on what? (a lock file? a source tree?)